runner/master_log.py

```diff
@@ -48,62 +67,50 @@ def end_test(self, test_id: str, timestamp: datetime):
 
     def add_event(self, event: LogEvent):
         self.log.append(event)
+        self.save_event(event)
+
+    def save_event(self, event: LogEvent):
+        with open(self.save_file, "a") as fd:
+            fd.write(json.dumps(event.to_json()) + "\n")
 
     def human_readable_full_log(self, test_id: str, message: str) -> List[str]:
         # Collate all the messages from the index point
         messages = []
         index = self.find_message(test_id, message)
 
         for event in self.log[index:]:
-            if event.type in [EventType.SEND_MESSAGE, EventType.RESPONSE_MESSAGE, EventType.SEND_FILL, EventType.RESPONSE_FILL]:
+            if event.type in [EventType.SEND_MESSAGE, EventType.RESPONSE_MESSAGE]:
                 sender = "Test" if event.type == EventType.SEND_MESSAGE else "Agent"
                 messages.append(f"{sender} ({event.timestamp}): {event.data['message']}")
+            elif event.type in [ EventType.SEND_FILL, EventType.RESPONSE_FILL]:
+                sender = "System" if event.type == EventType.SEND_FILL else "Agent"
+                messages.append(f"{sender} ({event.timestamp}): {event.data['message']}")
```

Comment:
This came to my mind while doing the review. Just a suggestion, in case you like it.
```python
sender = {
  EventType.SEND_MESSAGE: "Test",
  EventType.SEND_FILL: "System",
}.get(event.type, "Agent")
```